charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml

```diff
@@ -8,91 +8,114 @@ data:
+        name: {{ .Values.postgresql.auth.database }}
+        password: {{ .Values.config.db.sql.password }}
+        port: {{ .Values.config.db.sql.port }}
+        replica: {{ .Values.config.db.sql.replica }}
+        skip_initialize_with_version: {{ .Values.config.db.sql.skip_initialize_with_version }}
+        table_prefix: {{ .Values.config.db.sql.table_prefix }}
+        time_zone: {{ .Values.config.db.sql.time_zone }}
+        tx_timeout: {{ .Values.config.db.sql.tx_timeout }}
+        user: {{ .Values.postgresql.auth.username }}
+      {{- else if eq .Values.config.db.datastore.engine "sqlite" }}
+      sqlite:
+        database_path: {{ .Values.config.db.sqlite.database_path }}
+        debug: {{ .Values.config.db.sqlite.debug }}
+        max_connection_idle_time: {{ .Values.config.db.sqlite.max_connection_idle_time }}
+        max_connection_time: {{ .Values.config.db.sqlite.max_connection_time }}
+        max_idle_connections: {{ .Values.config.db.sqlite.max_idle_connections }}
+        max_open_connections: {{ .Values.config.db.sqlite.max_open_connections }}
+        shared: {{ .Values.config.db.sqlite.shared }}
+      {{- else }}
+      error: "Unsupported database engine specified. Please use 'postgresql' or 'sqlite'."
+      {{- end }}
+    debug_profiling: {{ .Values.config.debug_profiling }}
+    disable_itc: {{ .Values.config.disable_itc }}
+    arc:
+      url: {{ .Values.config.arc.url }}
+      token: {{ .Values.config.arc.token }}
+      {{- if .Values.config.arc.deployment_id }}
+        deployment_id: {{ .Values.config.arc.deployment_id }}
+      {{- else }}
+        deployment_id: 'spv-wallet-{{ printf "%s-$s"  (include "spv-wallet.fullname" .) .Values.global.domainName | adler32sum }}'
+      {{- end }}
       callback:
-        callback_host: 'https://{{.Values.global.domainName}}'
-      use_fee_quotes: {{ .Values.config.nodes.use_fee_quotes }}
+        enabled: {{ .Values.config.arc.callback.enabled }}
+        host: 'https://{{.Values.global.domainName}}'
+        {{- if .Values.config.arc.callback.token }}
+          token: {{ .Values.config.arc.callback.token }}
+        {{- end }}
+      use_fee_quotes: {{ .Values.config.arc.use_fee_quotes }}
       fee_unit:
-        satoshis: {{ .Values.config.nodes.fee_unit.satoshis }}
-        bytes: {{ .Values.config.nodes.fee_unit.bytes }}
+        satoshis: {{ .Values.config.arc.fee_unit.satoshis }}
+        bytes: {{ .Values.config.arc.fee_unit.bytes }}
     notifications:
       enabled: {{ .Values.config.notifications.enabled }}
-      webhook_endpoint: "{{ .Values.config.notifications.webhook_endpoint }}"
+    block_headers_service:
+      auth_token: ""
+      {{- $defaultUrl := printf "http://%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" -}}
+      {{- $blockHeadersServiceUrl := default $defaultUrl .Values.config.block_headers_service.url -}}
+      url: {{ $blockHeadersServiceUrl }}
```

Comment:
url property is generated in the same line as auth_token (probably because you don't need to use `-}}` but simply `}}` )